src/ucp/rma/flush.c

```diff
@@ -680,24 +686,51 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_ep_flush, (ep), ucp_ep_h ep)
 
 UCS_PROFILE_FUNC(ucs_status_t, ucp_worker_fence, (worker), ucp_worker_h worker)
 {
+    ucp_context_t *context = worker->context;
+    ucp_request_t *request;
     ucp_rsc_index_t rsc_index;
     ucp_worker_iface_t *wiface;
+    ucs_status_ptr_t status_ptr;
     ucs_status_t status;
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
-    UCS_BITMAP_FOR_EACH_BIT(worker->context->tl_bitmap, rsc_index) {
-        wiface = ucp_worker_iface(worker, rsc_index);
-        if (wiface->iface == NULL) {
-            continue;
-        }
+    if (!context->config.worker_strong_fence) {
```

Comment:
1. switch if/else
2. maybe separate to 2 helper functions? since they're doing 2 completely different things